contracts/MechMarketplace.sol

```diff
@@ -421,17 +467,85 @@ contract MechMarketplace {
         numUndeliveredRequests--;
         // Increase the amount of requester delivered requests
         mapDeliveryCounts[requester]++;
+        // Increase the amount of agent mech delivery counts
+        mapAgentMechDeliveryCounts[msg.sender]++;
         // Increase the amount of mech service multisig delivered requests
-        mapMechServiceDeliveryCounts[mechService]++;
+        mapMechServiceDeliveryCounts[mechServiceMultisig]++;
 
         // Increase mech karma that delivers the request
-        IKarma(karmaProxy).changeMechKarma(msg.sender, 1);
+        IKarma(karma).changeMechKarma(msg.sender, 1);
 
         emit MarketplaceDeliver(priorityMech, msg.sender, requester, requestId, requestData);
 
         _locked = 1;
     }
 
```

Comment:
Fees:

-> Request:
- check supplied native gas token matches mech's payment requirements >=

-> Deliver:
- we record who delivered; update mapping to account for full payments collected by a given mech

-> withdrawPayments()  (called by a mech)
- pulls all payments by given mech, less fee

-> drain as per design